packages/edit-site/src/components/sidebar-edit-mode/page-panels/page-slug.js

```diff
@@ -0,0 +1,159 @@
+/**
+ * WordPress dependencies
+ */
+import { useSelect, useDispatch } from '@wordpress/data';
+import { safeDecodeURIComponent, filterURLForDisplay } from '@wordpress/url';
+import { useState, useMemo } from '@wordpress/element';
+import { __experimentalInspectorPopoverHeader as InspectorPopoverHeader } from '@wordpress/block-editor';
+import { __ } from '@wordpress/i18n';
+import {
+	TextControl,
+	ExternalLink,
+	__experimentalHStack as HStack,
+	__experimentalVStack as VStack,
+	__experimentalText as Text,
+	Dropdown,
+	Button,
+} from '@wordpress/components';
+import { useEntityRecord, store as coreStore } from '@wordpress/core-data';
+import { store as noticesStore } from '@wordpress/notices';
+
+/**
+ * Internal dependencies
+ */
+import useEditedEntityRecord from '../../use-edited-entity-record';
+
+export const PERMALINK_POSTNAME_REGEX = /%(?:postname|pagename)%/;
+
+export default function PageSlug( { postType, postId } ) {
+	const { createErrorNotice } = useDispatch( noticesStore );
+	const { editEntityRecord } = useDispatch( coreStore );
+	const { record, isLoaded } = useEditedEntityRecord( postType, postId );
+	const { record: savedRecord } = useEntityRecord( postType, postId );
```

Comment:
Do we need all these data? I think selecting only what's needed is good Redux/WP data practice.

Reply:
I updated to get only the slug.

Reply:
This comment was regarding both hooks. Any reason for not using a similar selector as we have in `PostURL`?

I want to clarify that this is a personal nitpick, so feel free to ignore it. Don't like using general hooks when a component only needs one or two pieces of data.

P.S. Btw `useEntityRecord` returns all kinds of variations + methods for a record.

https://github.com/WordPress/gutenberg/blob/b483df2fd37d2d4c925f4e78e62512e9f0146fb2/packages/core-data/src/hooks/use-entity-record.ts#L202-L209

Reply:
Yeah, you're right, no need for all these. I used `getEditedEntityRecord`.